### PR TITLE
Add interfaces for incoming notifications

### PR DESCRIPTION
**Pull request by sergej-koscejev**
This is the companion pull request to thephpleague/omnipay#215.

### CODE REVIEW

**Comment by kayladnls** [comment]
I like the idea of adding interfaces here, but can't merge a failing build.

File: src/Omnipay/Common/Message/NotificationInterface.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+     * Gateway Reference
+     *
+     * @return string A reference provided by the gateway to represent this transaction
+     */
+    public function getTransactionReference();
+
+    /**
+     * Was the transaction successful?
+     *
+     * @return string Transaction status, one of {@see STATUS_COMPLETED}, {@see #STATUS_PENDING}, or {@see #STATUS_FAILED}.
```
Comment (barryvdh) [inline src/Omnipay/Common/Message/NotificationInterface.php]: This line is failing the codestyle checks. It's too long.

**Comment by sergej-koscejev** [comment]
All checks pass now.

**Comment by beinbm** [comment]
can someone merge it? i would love to base my next changes on the checkout.com gateway on it :)

**Comment by kayladnls** [comment]
:+1: Thank you @sergej-koscejev !

**Comment by beinbm** [comment]
thank you both!

**Pull request merged**